ios/Murcielago/Murcielago/UserInterface/Views/MainView/MainViewModel.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  MainViewModel.swift
+//  Murcielago
+//
+//  Created by Sam Franusic on 11/12/24.
+//
+
+import SwiftUI
+
+class MainViewModel: ObservableObject {
+    let authenticator = AuthenticationService()
+
+    @Published var unauthenticated: Bool = true
+    @Published var errorMessage: String = ""
+    @Published var serviceActivity = false
+
+    @MainActor
+    func authenticate(username: String, password: String) {
+        Task {
+            serviceActivity = true
+            clearErrorMessage()
+            async let result = authenticator.signIn(username: username, password: password)
+
+            if await result == false {
+                displayErrorMessage(type: .invalidCredentials)
+            }
+            await updateAuthentication()
+            serviceActivity = false
```

Comment:
Interesting, out of curiosity, why are you adding an authenticator? Is it necessary for this project?

Reply:
Just setting up a placeholder for now. Eventually will add a real authentication model.